### PR TITLE
make custom mir ICE a bit nicer

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_mir_build/src/build/custom/mod.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ fn parse_attribute(attr: &Attribute) -> MirPhase {
                 phase = Some(value);
             }
             other => {
-                panic!("Unexpected key {}", other);
+                bug!("Unexpected key while parsing custom_mir attribute: '{}'", other);
```
Comment (compiler-errors) [inline compiler/rustc_mir_build/src/build/custom/mod.rs:121]:
```suggestion
                span_bug!(nested.span, "Unexpected key while parsing custom_mir attribute: '{}'", other);
```

I think this works?

**Review by compiler-errors — APPROVED** [review]
r=me preferably with change. i think there's some span you can use here.

File: compiler/rustc_mir_build/src/build/custom/mod.rs (hunk 1 above)
Comment (matthiaskrgr) [reply compiler/rustc_mir_build/src/build/custom/mod.rs]: ah, neat! :)
mir_ice

**Comment by matthiaskrgr** [comment]
o.O

**Pull request merged**